go/proxyd/handle.go

```diff
@@ -0,0 +1,146 @@
+package proxyd
+
```

Comment:
This file contains the main logic that was previously in `server.HandleRPC` so that it can be reused, the difference is that `doRequest` is now passed in to send the request